multimodal/src/autogluon/multimodal/models/utils.py

```diff
@@ -686,28 +686,45 @@ def update_mmdet_config(key, value, config):
                     update_mmdet_config(key, value, subsubconfig)
 
 
-def run_model(model: nn.Module, batch: dict):
+def run_model(model: nn.Module, batch: dict, trt_model: Optional[nn.Module] = None):
+    from ..utils.onnx import OnnxModule
     from .document_transformer import DocumentTransformer
+    from .fusion.fusion_mlp import MultimodalFusionMLP
     from .huggingface_text import HFAutoModelForTextPrediction
+    from .t_few import TFewModel
     from .timm_image import TimmAutoModelForImagePrediction
 
-    supported_models = (TimmAutoModelForImagePrediction, HFAutoModelForTextPrediction)
+    supported_models = (
+        TimmAutoModelForImagePrediction,
+        HFAutoModelForTextPrediction,
+        MultimodalFusionMLP,
+        TFewModel,
+        OnnxModule,
+    )
     pure_model = model.module if isinstance(model, nn.DataParallel) else model
+    if isinstance(pure_model, OnnxModule):
```

Comment:
In which case `pure_model ` can be a `OnnxModule ` object? I don't see where `run_model` is called with a `OnnxModule ` object as input.

Reply:
For predicting with the TRT for a fusion model,
```python
onnx_module = predictor.export_tensorrt(data=X)
predictor._model = onnx_module
predictor.predict(X)  # pure_model would be OnnxModule here
```